api/grpc_disruption_test.go

```diff
@@ -0,0 +1,345 @@
+					ErrorToReturn:    "",
+					OverrideToReturn: "",
+					QueryPercent:     100,
+				},
+			},
+		}
+
+		err := spec.Validate()
+		Expect(err.Error()).To(Equal("the gRPC disruption must have either ErrorToReturn or OverrideToReturn specified for endpoint /chaos_dogfood.ChaosDogfood/order"))
+	})
+
+	It("Query percent of 100 does validate", func() {
+		spec := v1beta1.GRPCDisruptionSpec{
+			Port: 50051,
+			Endpoints: []v1beta1.EndpointAlteration{
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "",
+					QueryPercent:     100,
+				},
+			},
+		}
+
+		Expect(spec.Validate()).To(BeNil())
+	})
+
+	It("Query percent of 99 does validate", func() {
+		spec := v1beta1.GRPCDisruptionSpec{
+			Port: 50051,
+			Endpoints: []v1beta1.EndpointAlteration{
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "",
+					QueryPercent:     99,
+				},
+			},
+		}
+
+		Expect(spec.Validate()).To(BeNil())
+	})
+
+	It("Query percent of 100 on two endpoints does validate", func() {
+		spec := v1beta1.GRPCDisruptionSpec{
+			Port: 50051,
+			Endpoints: []v1beta1.EndpointAlteration{
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "",
+					QueryPercent:     100,
+				},
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/getCatalog",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "",
+					QueryPercent:     100,
+				},
+			}}
+
+		Expect(spec.Validate()).To(BeNil())
+	})
+
+	It("Query percent which sums to 100 on two endpoints does validate", func() {
+		spec := v1beta1.GRPCDisruptionSpec{
+			Port: 50051,
+			Endpoints: []v1beta1.EndpointAlteration{
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "",
+					QueryPercent:     60,
+				},
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "ALREADY_EXISTS",
+					OverrideToReturn: "",
+					QueryPercent:     40,
+				},
+			}}
+
+		Expect(spec.Validate()).To(BeNil())
+	})
+
+	It("Query percent which sums to less than 100 on two endpoints does validate", func() {
+		spec := v1beta1.GRPCDisruptionSpec{
+			Port: 50051,
+			Endpoints: []v1beta1.EndpointAlteration{
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "CANCELED",
+					OverrideToReturn: "",
+					QueryPercent:     49,
+				},
+				{
+					TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+					ErrorToReturn:    "ALREADY_EXISTS",
+					OverrideToReturn: "",
+					QueryPercent:     49,
+				},
+			}}
+
+		Expect(spec.Validate()).To(BeNil())
+	})
+
+	It("Query percent which sums to more than 100 on two endpoints does not validate", func() {
+		spec := v1beta1.GRPCDisruptionSpec{
+			Port: 50051,
```

Comment:
You could use a `BeforeEach` block to create a spec object and define the Port in only one place, so you dont have to repeat this line.